### PR TITLE
Support ItemID

### PR DESCRIPTION
**Pull request by Roms1383**
This PR adds support for ItemID, along with tests.

### CODE REVIEW

**Comment by Roms1383** [comment]
Side note: in this implementation, as opposed to `red4ext-rs`, I implemented the enum conversion as infallible (`From`) based on the assumptions that `Invalid` and `None` respectively convey the idea that the enums are somehow invalid anyway. If you prefer the `TryFrom` version, I can revert these changes back @jac3km4.

**Comment by Roms1383** [comment]
One thing that I would personally find super ergonomic would be to encode directly in the type whether the `ItemID` is `query` or non-`query`, since it's something that is very often overlooked. But yesterday when making tests with Redscript to check what differentiate both in terms of their internal values, I couldn't figure it out. I think that I can find it out by creating them in Redscript and logging their fields with `red4ext-rs` but it was late and I left it aside. What is your opinion on this @jac3km4 ?

File: src/types/item_id.rs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+impl From<u8> for GameEItemIdFlag {
+    fn from(value: u8) -> Self {
+        if value == 1 {
+            return Self::Preview;
+        }
+        Self::None
+    }
+}
+
+impl ItemId {
```
Comment (jac3km4) [inline src/types/item_id.rs:63]: move this impl below the struct

File: src/types/item_id.rs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+impl Default for Seed {
+    fn default() -> Self {
+        Self(DEFAULT_ITEM_ID_RNG_SEED)
+    }
+}
+
+#[derive(Debug, Default, Clone, Copy, PartialEq, PartialOrd, Eq)]
+#[repr(u8)]
+#[allow(dead_code)]
```
Comment (jac3km4) [inline src/types/item_id.rs:21]: why is allow(dead_code) needed here?

File: src/types/tweak_db_id.rs [hunk 3]
```diff
@@ -64,6 +64,18 @@ impl TweakDbId {
...
+        unsafe { self.0.__bindgen_anon_1.name }.hash
+    }
+
+    pub(super) const fn len(&self) -> u8 {
+        unsafe { self.0.__bindgen_anon_1.name }.length
+    }
+
+    pub(super) const fn into_inner(self) -> red::TweakDBID {
+        self.0
+    }
```
Comment (jac3km4) [inline src/types/tweak_db_id.rs:78]: `Copy` types shouldn't use `into_` in methos names according to [Rust API guidelines](https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv)
```suggestion
    pub(super) const fn to_inner(self) -> red::TweakDBID {
        self.0
    }
```

**Comment by jac3km4** [comment]
> One thing that I would personally find super ergonomic would be to encode directly in the type whether the ItemID is query or non-query
I'm fine with that as long as the type remains `#[repr(transparent)]`,
you could support it by adding another wrapper type or a type argument to `ItemId`

File: src/types/item_id.rs (hunk 1 above)
Comment (Roms1383) [reply src/types/item_id.rs]: moved in 2c2ad8b

File: src/types/item_id.rs (hunk 2 above)
Comment (Roms1383) [reply src/types/item_id.rs]: fixed in 5dd0cb8

File: src/types/tweak_db_id.rs (hunk 3 above)
Comment (Roms1383) [reply src/types/tweak_db_id.rs]: renamed in 0ec8365

File: src/types/item_id.rs [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    pub fn structure(&self) -> GamedataItemStructure {
+        self.0.structure.into()
+    }
+
+    pub fn flags(&self) -> GameEItemIdFlag {
+        self.0.flags.into()
+    }
+
+    #[inline]
+    pub fn get_tdbid(&self) -> TweakDbId {
```
Comment (jac3km4) [inline src/types/item_id.rs:31]: I'd rename it for consistency, the other methods are also getters but aren't prefixed with `get_`
```suggestion
    pub fn tdbid(&self) -> TweakDbId {
```

Comment (Roms1383) [reply src/types/item_id.rs]: renamed in 413fee2

**Pull request merged**